### PR TITLE
New Docs Content: Button alignment, order, and grouping

### PR DESCRIPTION
**Pull request by jorytindall**
### :pushpin: Summary

If merged this PR will migrate, update, and add to the documentation, examples, and guidance around button alignment, grouping, and order as captured in [this research](https://docs.google.com/document/d/1bm5loyxU-SjCAJlPjX9pT6CLAEAqG_jtbVXUis2NfY8/edit?usp=sharing) and [this presentation](https://docs.google.com/presentation/d/16ywGl_AE5YUEpqQOFrExNoyZ4vkDdX-HMGwANGQ8PNs/edit?usp=sharing)

### :hammer_and_wrench: Detailed description

Some of this content is still in draft, but I think at this time it's ready for an initial pass. This is _a lot_ of content, here's a bit about the narrative I'm trying to express:

- Overview about some of the high-level UX concepts (reading patterns, constrained vs unconstrained, etc)
- Details about each strategy and how they map back to the UX concepts
    - Alignment
    - Order
    - Grouping
- Each one of these sections has examples of do/dont as well as the impact behind each strategy

### Still to do

- There are a few more exceptions I'd like to work through, but I think rather than putting them in their own section I'll weave them into the rest of the content.
- The accessibility page is currently empty, I've been keeping a11y points inline with the topic they relate to, so this might not be necessary
- The title of the pattern still needs some work, maybe "Button organization"?

### :link: External links

Jira ticket: [HDS-1244](https://hashicorp.atlassian.net/browse/HDS-1244)

[HDS-1244]: https://hashicorp.atlassian.net/browse/HDS-1244?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+While the alignment, order, and grouping of buttons and actions might seem trivial, the impact on users is immediate and often occurs in experiences of high important and interruption.
+
+These guidelines don't cover every scenario and are intended to aid in the decision-making process when determining how to organize buttons and actions. Different methods of alignment and ordering will result in different outcomes with pros and cons for each. What is most important is **consistency** across an application in how actions are aligned, ordered and grouped.
+
+!!! Info
+
+Button alignment isn't relevant for every component or pattern, but is rather aimed at functional elements; i.e. a modal, forms, and page-level actions.
```
Comment (MelSumner) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:7]:
```suggestion
Button alignment isn't relevant for every component or pattern, but is rather aimed at functional elements; i.e. modals, forms, and page-level actions.
```

Suggestion: pluralize modal for consistency with the other items in the group.

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+All three of these reading patterns are relevant and useful when observed in isolation, however, the reality of how users read, consume, and comprehend content is likely a combination of multiple reading patterns that change and overlap depending on the context. With that in mind, understanding these patterns and how they are relevant to different types of UX content will help to produce the best experience.
+
+## Key concepts
+
+- No one layout or reading pattern is perfect, users will generally shift between different patterns depending on the content.
+- Adherence to specific reading pattern **is not** an excuse for lack of hierarchy within a layout.
+    - Introducing hierarchical focal points using typography and color in tandem with reading patterns is often the most effective method to communicate the purpose and function of a layout, pattern, or component.
+- There is a fair amount of overlap between each pattern.
+
+
```
Comment (MelSumner) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:100]:
```suggestion
```

File: website/docs/patterns/button-alignment/partials/specifications/alignment.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+!!! Do
+
+In most scenarios, align buttons to the **left side** of a container or page.
+
+- This reinforces nautral reading patterns and a conversational "dialog-like" messaging strategy.
+- When combined with a consistent ordering method, the primary button is prioritized in the reading direction.
+- When considering different contexts, opting for a single method helps to create consistency between constrained and unconstrained components and patterns.
+!!!
+
+![Button alignment reading pattern](/assets/patterns/button-alignment/alignment-reading-pattern.png)
```
Comment (MelSumner) [inline website/docs/patterns/button-alignment/partials/specifications/alignment.md:26]:
```suggestion
![Button alignment reading pattern](/assets/patterns/button-alignment/alignment-reading-pattern.png)

```

File: website/docs/patterns/button-alignment/partials/specifications/exceptions.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+## Exceptions
+
+- Actions at the page level
+- Actions in a progressive, multi-step flow
+
+**Still working on this section**
```
Comment (MelSumner) [inline website/docs/patterns/button-alignment/partials/specifications/exceptions.md:6]:
```suggestion
**Still working on this section**

```

File: website/docs/patterns/button-alignment/partials/specifications/grouping.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+- Group buttons together that perform similar, related, and inverse functions. This generally refers to functions like "yes/no" or "ok/cancel." 
+- Inverse grouping generally helps users to make decisions quicker as their options are limited.
+- As a best practice, tertiary actions should only be used if they help the user make a decision (linking to a different area of the application or documentation).
+    - Presenting the user with more than one action in this context can be confusing and force the user to question whether they are making the correct decision. Read more about [Hick's Law](https://www.interaction-design.org/literature/article/hick-s-law-making-the-choice-easier-for-users).
+
+!!! Do
+
+![Grouping inverse actions](/assets/patterns/button-alignment/grouping-inverse-actions.png)
+!!!
```
Comment (MelSumner) [inline website/docs/patterns/button-alignment/partials/specifications/grouping.md:13]:
```suggestion
!!!

```

File: website/docs/patterns/button-alignment/partials/specifications/order.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+- User edits the settings of a cluster
+    - "Would you like to learn more about this action?" "No"
+    - "Would you like to cancel?" "No"
+    - "Would you like to save the changes to this cluster?" "Yes, finally that's what I was trying to do all along."
+!!!
+
+!!! Do
+
+#### Nautral conversation based on priority
```
Comment (MelSumner) [inline website/docs/patterns/button-alignment/partials/specifications/order.md:47]:
```suggestion
#### Natural conversation based on priority
```

**Review by MelSumner — APPROVED** [review]
A few suggestions but otherwise :shipit:

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
+While the alignment, order, and grouping of buttons and actions might seem trivial, the impact on users is immediate and often occurs in experiences of high important and interruption.
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:1]: Nit: This sentence seems a bit hard to parse. Are there ways it could be broken up or clarified?

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+These guidelines don't cover every scenario and are intended to aid in the decision-making process when determining how to organize buttons and actions. Different methods of alignment and ordering will result in different outcomes with pros and cons for each. What is most important is **consistency** across an application in how actions are aligned, ordered and grouped.
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:3]:
```suggestion
While these guidelines don’t cover every scenario, they are intended to aid in the decision-making process when determining how to organize buttons and actions. Different methods of alignment and ordering will result in different outcomes with pros and cons for each. What is most important is **consistency** across an application in how actions are aligned, ordered, and grouped.
```

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+!!! Info
+
+Button alignment isn't relevant for every component or pattern, but is rather aimed at functional elements; i.e. modals, forms, and page-level actions.
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:7]: I'm not sure if this is helpful information? Questions of buttons alignment would naturally only apply to cases in which you need to use buttons.

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+- **Unconstrained**
+    - General page content
+    - In-page forms
+    - The "window" in a general sense
+
+This differentiation is important because the expectations from the user are different in each context. A constrained component or pattern will often be used in an interstitial or circumstantial experience; i.e. an experience that is triggered by the user or the application illiciting a quick, progressive action.
+
+Unconstrained generally refers to a more "static" experience that isn't triggered by a user action. This is often tied to a page-level function or a navigational element.
+
+**Insert some supporting images here**
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:28]: For a note like this, you could put a TODO comment instead so that it wouldn't be visible to people visiting the site in case it gets committed.

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+## Layout and reading patterns
+
+When users read content, the pattern their eyes follow generally adheres to one or more reading pattern. By identifying and familiarizing ourselves  with these patterns we can better understand how the layout within our application, pattern, or component impacts comprehension and interpretation of the content.
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:32]:
```suggestion
When users read content, the pattern their eyes follow generally adheres to one or more reading patterns. By identifying and familiarizing ourselves with these patterns we can better understand how the layout within our application, pattern, or component impacts comprehension and interpretation of the content.
```

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+### Gutenberg diagram
+
+The Gutenberg diagram refers to a pattern the eyes follow when moving or scanning through **evenly distributed, homogenous content.** This pattern is most relevant in text-heavy scenarios like novels and newspapers, but is relevant in a UX context depending on the complexity of the content.
+
+![Visual example of a Gutenberg diagram](/assets/patterns/button-alignment/gutenberg-diagram.png)
+
+#### Characteristics
+
+- The users eyes progress through the content on an **axis of orientation** which moves further away from the start edge on each line or break in the content
+- Content that falls into the weak fallow area recieves less attention
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:49]:
```suggestion
- Content that falls into the weak fallow area receives less attention
```

Comment (jorytindall) [reply website/docs/patterns/button-alignment/partials/overview/overview.md:49]: That's a good point, I don't think it's an often used word, but it's supported by the graphic. That's generally the language that is used when referring to this specific reading pattern.

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+- This pattern indicates that important content and elements should be placed upon the **reading gravity path**
+- This pattern is most prevalent in content with little to no typographic hierarchy (i.e. long-form paragraphs)
+- This pattern is considered "in harmony" with natural reading direction
+
+### Z-shaped pattern
+
+A Z-shaped pattern refers to a pattern that follows the shape of the letter Z:
+
+1. The users eyes start in the top left and move horizontally to the right
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:56]:
```suggestion
1. The user’s eyes start in the top left and move horizontally to the right
```

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+2. Upon reaching the end (or terminal) of the line, the eyes move diagonally to the next line
+3. The readers eyes end at the bottom right of the content
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:58]:
```suggestion
3. The reader’s eyes end at the bottom right of the content
```

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+
+![Visual example of a Z-shaped pattern](/assets/patterns/button-alignment/z-shaped-pattern.png)
+
+#### Characteristics
+
+- There are _technically_ no fallow areas (areas that recieve less attention or lack comprehension) as in the Gutenberg pattern, though this is not always the case in practice.
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:64]: Maybe this definition of "fallow" could be referenced where it's used above too if you keep the term there.

Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:64]:
```suggestion
- There are _technically_ no fallow areas (areas that receive less attention or lack comprehension) as in the Gutenberg pattern, though this is not always the case in practice.
```

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+- This type of pattern is most prevalent in task-based functions or sequences of elements; i.e. a form or series of onboarding steps.
+- This pattern assumes that users read through _all_ of the content sequentially.
+
+### F-shaped pattern
+
+An F-shaped pattern loosely follows the shape of the letter F, first identified through eye-tracking and heatmap studies by the [Neilsen/Norman Group](https://www.nngroup.com/articles/f-shaped-pattern-reading-web-content-discovered/).
+
+![Visual example of an F-shaped pattern](/assets/patterns/button-alignment/f-shaped-pattern.png)
+
+**We probably can't use this image directly on our webiste :(**
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:74]: Maybe we should remove it to be safe? I think as long as we link to the page on their site which explains this pattern then that's probably fine. (Or we could make our own image somehow.)

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+
+![An example of eye tracking and heatmaps from the NN/group](/assets/patterns/button-alignment/f-shaped-pattern-nn-group.jpeg)
+
+#### Characteristics
+
+The F-shaped pattern is a prominent example of how users scan through content and how most users don't read content completely.
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:80]:
```suggestion
The F-shaped pattern is a prominent example of how users scan through content and how most users don’t read content completely.
```

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+    - Alerts
+    - Banners
+- **Unconstrained**
+    - General page content
+    - In-page forms
+    - The "window" in a general sense
+
+This differentiation is important because the expectations from the user are different in each context. A constrained component or pattern will often be used in an interstitial or circumstantial experience; i.e. an experience that is triggered by the user or the application illiciting a quick, progressive action.
+
+Unconstrained generally refers to a more "static" experience that isn't triggered by a user action. This is often tied to a page-level function or a navigational element.
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:21]:
```suggestion
Unconstrained generally refers to a more "static" experience that isn’t triggered by a user action. This is often tied to a page-level function or a navigational element.
```

File: website/docs/patterns/button-alignment/partials/overview/overview.md (hunk 16 above)
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:74]:
```suggestion
**We probably can’t use this image directly on our website :(**
```

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+- Information of lesser importance should be concise, potentially bulletted, and aligned to the left hand side of the page, pattern, or component.
+- This layout assumes users only scan content, rather than reading word-for-word.
+
+### Reading patterns in practice
+
+All three of these reading patterns are relevant and useful when observed in isolation, however, the reality of how users read, consume, and comprehend content is likely a combination of multiple reading patterns that change and overlap depending on the context. With that in mind, understanding these patterns and how they are relevant to different types of UX content will help to produce the best experience.
+
+## Key concepts
+
+- No one layout or reading pattern is perfect, users will generally shift between different patterns depending on the content.
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:93]:
```suggestion
- No particular layout or reading pattern is perfect, users will generally shift between different patterns depending on the content.
```

Comment (KristinLBradley) [reply website/docs/patterns/button-alignment/partials/overview/overview.md:93]: "one" doesn't seem quite clear here although there are other word choices than I suggested if you think there's a better option.

File: website/docs/patterns/button-alignment/partials/specifications/alignment.md [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+When considering internationalization (i18n), **start** and **end** are suitable substitutes for **left** and **right** alignment to account for languages that are read right-to-left ([RTL](https://developer.mozilla.org/en-US/docs/Glossary/RTL)).
+!!!
+
+![Button alignment options](/assets/patterns/button-alignment/alignment-methods.png)
+
+!!! Do
+
+In most scenarios, align buttons to the **left side** of a container or page.
+
+- This reinforces nautral reading patterns and a conversational "dialog-like" messaging strategy.
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/specifications/alignment.md:20]:
```suggestion
- This reinforces natural reading patterns and a conversational "dialog-like" messaging strategy.
```

File: website/docs/patterns/button-alignment/partials/specifications/order.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
+## Order
+
+When performing an action or function, users generally think from highest to lowest priority and from positive to negative. How buttons and actions are ordered should respect this by ordering the primary (most important) action first, secondary action second, and if applicable, tertiary action last.
+
+!!! Do
+
+Order buttons based on priority starting with the **highest** priority  action(usually by using a primary button) to **lowest** priority (either secondary or tertiary actions).
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/specifications/order.md:7]:
```suggestion
Order buttons based on priority starting with the **highest** priority action (usually by using a primary button) to **lowest** priority (either secondary or tertiary actions).
```

File: website/docs/patterns/button-alignment/partials/specifications/order.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+!!! Dont
+
+#### Non-natural conversation based on priority
+
+![Suboptimal grouping](/assets/patterns/button-alignment/suboptimal-grouping.png)
+
+- User edits the settings of a cluster
+    - "Would you like to learn more about this action?" "No"
+    - "Would you like to cancel?" "No"
+    - "Would you like to save the changes to this cluster?" "Yes, finally that's what I was trying to do all along."
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/specifications/order.md:42]:
```suggestion
    - "Would you like to save the changes to this cluster?" "Yes, finally that’s what I was trying to do all along."
```

File: website/docs/patterns/button-alignment/partials/specifications/order.md [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+#### Natural conversation based on priority
+![Optimal order](/assets/patterns/button-alignment/optimal-ordering.png)
+
+- User edits the settings of a cluster
+    - "Would you like to save the changes to this cluster?" "Yes, thank you."
+!!!
+
+#### Exceptions
+
+An possible exception to this ordering method is in multi-step and progressive flows in which buttons fulfill both a functional purpose (create or edit something) as well as a navigational purpose (proceed, go back).
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/specifications/order.md:56]:
```suggestion
A possible exception to this ordering method is in multi-step and progressive flows in which buttons fulfill both a functional purpose (create or edit something) as well as a navigational purpose (proceed, go back).
```

**Review by KristinLBradley — APPROVED** [review]

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 24]
```diff
@@ -44,7 +44,7 @@ The Gutenberg diagram refers to a pattern the eyes follow when moving or scannin
 #### Characteristics
 
 - The users eyes progress through the content on an **axis of orientation** which moves further away from the start edge on each line or break in the content
-- Content that falls into the weak fallow area receives less attention
+- Content falls into strong and weak _fallow areas_ where it receives elss attention and lacks comprehension compared to content along the axis of orientation
```
Comment (KristinLBradley) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:47]:
```suggestion
- Content falls into strong and weak _fallow areas_ where it receives less attention and lacks comprehension compared to content along the axis of orientation
```
